### PR TITLE
Tenor gifs

### PR DESCRIPTION
**Pull request by Vashist-Agarwalla**

### CODE REVIEW

File: lib/src/client/models/gif.dart [hunk 1]
```diff
@@ -144,3 +145,133 @@ class GiphyGif {
...
+      'duration': duration,
+      'preview': preview,
+      'dims': dims,
+      'size': size,
+    };
+  }
+}
+
+GiphyGif convertTenorGifToGiphyGif(TenorGif tenorGif) {
+  GiphyGif giphyGif = GiphyGif(
```
Comment (nikb7) [inline lib/src/client/models/gif.dart]: final

Comment (nikb7) [reply lib/src/client/models/gif.dart]: return directly

Comment (nikb7) [reply lib/src/client/models/gif.dart]: don't declare it here

File: lib/src/client/models/gif.dart [hunk 2]
```diff
@@ -144,3 +145,133 @@ class GiphyGif {
...
+    sourceTld: null,
+    sourcePostUrl: null,
+    isSticker: null,
+    importDatetime: null,
+    trendingDatetime: null,
+    user: null,
+    images: GiphyImages(
+      fixedWidth: GiphyFullImage(
+        url: tenorGif.mediaFormats?.url ?? '',
+        width: tenorGif.mediaFormats?.dims![0].toString() ?? '200',
```
Comment (nikb7) [inline lib/src/client/models/gif.dart]: dims! - nullable?

File: lib/src/client/tenor_client.dart [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+          'media_filter': mediaFormat,
+          if (next != null) 'pos': next,
+        },
+      ),
+      mediaFormat,
+    );
+  }
+
+  Future<TenorCollection> search(String query,
+      {int limit = 30,
```
Comment (nikb7) [inline lib/src/client/tenor_client.dart]: reduce

File: lib/src/views/tab/tab_bottom.dart [hunk 4]
```diff
@@ -1,10 +1,24 @@
...
+  const TabBottom({
     Key? key,
+    required this.isGIF,
   }) : super(key: key);
 
+  @override
+  State<TabBottom> createState() => _TabBottomState();
+}
+
+class _TabBottomState extends State<TabBottom> {
```
Comment (nikb7) [inline lib/src/views/tab/tab_bottom.dart]: relook

File: lib/src/client/models/gif.dart [hunk 5]
```diff
@@ -144,3 +145,96 @@ class GiphyGif {
...
+  String? created;
+  String? contentDescription;
+  String? itemUrl;
+  String? url;
+  List<String> tags;
+  List<String> flags;
+  bool? hasAudio;
+
+  TenorGif({
+    required this.id,
```
Comment (smitbarmase) [inline lib/src/client/models/gif.dart]: can we use positional args here?

File: lib/src/client/models/gif.dart [hunk 6]
```diff
@@ -144,3 +145,96 @@ class GiphyGif {
...
+  MediaFormats? mediaFormats;
+  String? created;
+  String? contentDescription;
+  String? itemUrl;
+  String? url;
+  List<String> tags;
+  List<String> flags;
+  bool? hasAudio;
+
+  TenorGif({
```
Comment (nikb7) [inline lib/src/client/models/gif.dart]: const

File: lib/src/client/tenor_client.dart [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+      throw TenorClientError(response.statusCode, response.body);
+    }
+  }
+}
+
+class TenorClientError {
+  final int statusCode;
+  final String exception;
+
+  TenorClientError(this.statusCode, this.exception);
```
Comment (nikb7) [inline lib/src/client/tenor_client.dart]: const

File: lib/src/views/tab/tenor_tab_details.dart [hunk 8]
```diff
@@ -0,0 +1,293 @@
...
+      _aspectRatio = gif.mediaFormats!.dims![0] / gif.mediaFormats!.dims![1];
+    } else {
+      _aspectRatio = 1.0;
+    }
+
+    return ClipRRect(
+      borderRadius: BorderRadius.circular(10.0),
+      child: InkWell(
+        onTap: () => _selectedGif(gif),
+        child: gif.mediaFormats == null || gif.mediaFormats?.url == null
```
Comment (nikb7) [inline lib/src/views/tab/tenor_tab_details.dart]: redundant

File: lib/src/views/tab/tenor_tab_details.dart [hunk 9]
```diff
@@ -0,0 +1,293 @@
...
+            ? Container()
```
Comment (nikb7) [inline lib/src/views/tab/tenor_tab_details.dart]: SizedBox or container?

File: lib/src/views/tab/tenor_tab_details.dart [hunk 10]
```diff
@@ -0,0 +1,293 @@
...
+                gif.mediaFormats!.url!,
+                semanticLabel: gif.title,
+                cache: true,
+                gaplessPlayback: true,
+                fit: BoxFit.fill,
+                headers: {'accept': 'image/*'},
+                loadStateChanged: (state) => AnimatedSwitcher(
+                  duration: const Duration(milliseconds: 350),
+                  child: gif.mediaFormats == null
+                      ? Container()
```
Comment (nikb7) [inline lib/src/views/tab/tenor_tab_details.dart]: same

File: lib/src/views/tab/tenor_tab_details.dart [hunk 11]
```diff
@@ -0,0 +1,302 @@
...
+                ),
+              ),
+      ),
+    );
+  }
+
+  Future<void> _loadMore() async {
+    //Return if is loading or no more gifs
+    if (_isLoading) {
+      print("No more object");
```
Comment (nikb7) [inline lib/src/views/tab/tenor_tab_details.dart]: remove

**Pull request merged**